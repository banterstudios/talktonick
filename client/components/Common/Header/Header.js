import React, { Component } from 'react'
import Wrapper from 'client/components/Gui/Wrapper'
import glamorous from 'glamorous'
import { Grid, Row } from 'react-bootstrap'
import Navbar from 'client/components/Navigation/Navbar'
import NavLink from 'client/components/Navigation/NavLink'
import NavItem from 'client/components/Navigation/NavItem'
import Logo from 'client/components/Common/Logo'

const HeaderWrapper = glamorous.header(({ theme: { headerBgColor, headerHeight } }) => ({
  position: 'fixed',
  top: '0',
  left: '0',
  width: '100%',
  height: headerHeight,
  background: headerBgColor,
  zIndex: '99'
}))

const LogoWrapper = glamorous.div(() => ({
  position: 'absolute',
  top: '50%',
  left: '10px',
  transform: 'translateY(-50%)'
}))

const GridWrapper = glamorous.div(({ theme: { headerHeight } }) => ({
  position: 'relative',
  padding: `0 60px`
}))

class Header extends Component {
  render () {
    return (
      <HeaderWrapper>
        <Wrapper className='header'>
          <LogoWrapper>
            <Logo modifier='sm' />
          </LogoWrapper>
          <GridWrapper>
            <Grid>
              <Row>
                <Navbar>
                  <NavItem>
                    <NavLink href='/'>Home</NavLink>
                  </NavItem>
                  <NavItem>
                    <NavLink href='/about'>About</NavLink>
                  </NavItem>
                </Navbar>
              </Row>
            </Grid>
          </GridWrapper>
        </Wrapper>
      </HeaderWrapper>
    )
  }
}

export default Header
