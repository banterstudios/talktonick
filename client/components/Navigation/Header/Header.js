import React, { Component } from 'react'
import Wrapper from 'client/components/Gui/Wrapper'
import glamorous from 'glamorous'
import { Grid, Row, Col } from 'react-bootstrap'

const HeaderWrapper = glamorous.header(({ theme: { bgPrimary, headerHeight } }) => ({
  position: 'fixed',
  top: '0',
  left: '0',
  width: '100%',
  height: headerHeight,
  background: bgPrimary,
  zIndex: '99'
}))

class Header extends Component {
  render () {
    return (
      <HeaderWrapper>
        <Wrapper className='header'>
          <Grid>
            <Row>
              <Col xs={2} />
              <Col xs={8}>
                Talk To Nick
              </Col>
              <Col xs={2} />
            </Row>
          </Grid>
        </Wrapper>
      </HeaderWrapper>
    )
  }
}

export default Header
