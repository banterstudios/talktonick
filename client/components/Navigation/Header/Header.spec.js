import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Header from './Header'

describe('components/Navigation/Header', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = mount(<Header />)

      expect(toJson(component)).toMatchSnapshot()
      component.unmount()
    })
  })
})
