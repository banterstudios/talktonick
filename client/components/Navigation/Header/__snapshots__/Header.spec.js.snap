// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navigation/Header @renders should render correctly 1`] = `
<Header>
  <glamorous(header)>
    <header
      className="css-p4zpk5"
    >
      <glamorous(section)
        className="header"
      >
        <section
          className="css-4kgshc header"
        >
          <Grid
            bsClass="container"
            componentClass="div"
            fluid={false}
          >
            <div
              className="container"
            >
              <Row
                bsClass="row"
                componentClass="div"
              >
                <div
                  className="row"
                >
                  <Col
                    bsClass="col"
                    componentClass="div"
                    xs={2}
                  >
                    <div
                      className="col-xs-2"
                    />
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    xs={8}
                  >
                    <div
                      className="col-xs-8"
                    >
                      <Title
                        type="h1"
                      >
                        <glamorous(h1)>
                          <h1
                            className="css-jm9yv1"
                          >
                            Talk To Nick
                          </h1>
                        </glamorous(h1)>
                      </Title>
                    </div>
                  </Col>
                  <Col
                    bsClass="col"
                    componentClass="div"
                    xs={2}
                  >
                    <div
                      className="col-xs-2"
                    />
                  </Col>
                </div>
              </Row>
            </div>
          </Grid>
        </section>
      </glamorous(section)>
    </header>
  </glamorous(header)>
</Header>
`;
